src/Application/Actions/Donations/Confirm.php

```diff
@@ -238,9 +238,13 @@ protected function action(Request $request, Response $response, array $args): Re
             ], 500);
         }
 
-        $updatedIntent = $this->stripeClient->paymentIntents->retrieve($paymentIntentId);
-
+        // Assuming Stripe calls worked, commit any changes that `deriveFees()` made to the EM-tracked `$donation`.
         $this->entityManager->flush();
+        $this->entityManager->commit();
```

Comment:
Looks like adding `commit` will fix the immediate issue. Either here or as a rule in general maybe we should avoid using `beginTransaction` and always use `\Doctrine\ORM\EntityManagerInterface::transactional` instead since that automatically commits if no exception is thrown. Also our own implementation of that, `\MatchBot\Application\Persistence\RetrySafeEntityManager::transactional` auto retries the commit, but we don't have our own implementation of `\Doctrine\ORM\EntityManagerInterface::commit` so that one won't auto retry. It's a bit unintuitive that they behave differently.

Reply:
I'm not totally sure either way on this. I definitely prefer `transactional()` for simple things with one failure mode, but I think it became a bit unwieldy before in a method where we wanted to have a few distinct exception handle clauses which would each rollback, sometimes conditionally. I'm not sure whether Confirm is simple enough to be in the former category or not.

Reply:
This commit clearly is the line that fixes the bug.